core/src/common.cpp

```diff
@@ -253,6 +253,24 @@ namespace irods::http
 		log::trace("{}: Client is authenticated.", __func__);
 		return {.client_info = std::move(*client_info)};
 	} // resolve_client_identity
+
+	auto get_port_from_url(boost::urls::url_view _url) -> std::optional<std::string> {
+		if (_url.has_port()) {
+			return _url.port();
+		}
+
+		switch (auto scheme_type{_url.scheme_id()}; scheme_type) {
+		case boost::urls::scheme::https:
+			return "443";
+			break;
+		case boost::urls::scheme::http:
+			return "80";
+			break;
+		default:
+			log::error("{}: Cannot deduce port from url.", __func__);
```

Comment:
If possible, consider printing the URL in the error message as well? Perhaps it doesn't add much...

Reply:
Thought about that too, just wasn't sure how to get the full string out of `url_view`.

Think I could just `.data()` to get it. I'll try it out.